corehq/apps/case_importer/do_import.py

```diff
@@ -557,7 +557,8 @@ def add_error(self, row_num, error):
         if 'rows' not in self._errors[key][column_name]:
             self._errors[key][column_name]['rows'] = []
 
-        self._errors[key][column_name]['rows'].append(row_num)
+        if row_num not in self._errors[key][column_name]['rows']:
```

Comment:
What happens if a row has multiple, different types of errors?

Reply:
Then that row number would appear for each error, for example if you see row 2 in the screenshot in the description.